### PR TITLE
Add Xbox support for the XPlatCpp SDK

### PR DESCRIPTION
**Pull request by sanchitgarg**

### CODE REVIEW

File: targets/XPlatCppSdk/source/code/source/playfab/OneDSIXHR2Plugin.cpp [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        HttpRequest postEventRequest;
+
+        // Setup headers
+        std::vector<HttpHeaderInfo> headers;
+        SetupRequestHeaders(reqContainer, headers);
+
+        // Setup url
+        std::wstring url = L"https://self.events.data.microsoft.com/OneCollector/1.0/";
+
+        // Setup payload
```
Comment (sanchitgarg) [inline targets/XPlatCppSdk/source/code/source/playfab/OneDSIXHR2Plugin.cpp]: Remove

File: targets/XPlatCppSdk/source/code/source/playfab/PlayFabPluginManager.cpp [hunk 2]
```diff
@@ -77,6 +77,10 @@ namespace PlayFab
 
     std::shared_ptr<IPlayFabPlugin> PlayFabPluginManager::CreatePlayFabTransportPlugin()
     {
+#ifdef _DURANGO
+        return std::shared_ptr<PlayFabIXHR2HttpPlugin>(new PlayFabIXHR2HttpPlugin());
+#else
         return std::shared_ptr<PlayFabHttp>(new PlayFabHttp());
+#endif // !_DURANGO
```
Comment (sanchitgarg) [inline targets/XPlatCppSdk/source/code/source/playfab/PlayFabPluginManager.cpp]: !_DURANGO

File: targets/XPlatCppSdk/source/code/stdafx.h [hunk 3]
```diff
@@ -21,19 +21,19 @@
...
+#ifdef _DURANGO
+#endif // _DURANGO
+
+#ifdef __linux__
 #include <stdio.h>
-#endif
+#endif // __linux__
 
 #ifdef _WIN32
+#ifndef _DURANGO
```
Comment (sanchitgarg) [inline targets/XPlatCppSdk/source/code/stdafx.h]: Add a comment explaining why we need !_DURANGO

File: targets/XPlatCppSdk/source/.gitignore.ejs [hunk 4]
```diff
@@ -304,7 +307,24 @@ external/openssl-build-debug/**/*
...
+external/jsoncpp-xdk-build/Release
 
+# Linux Build
 build/Linux/*.o
-build/Linux/XPlatCppLinux
\ No newline at end of file
+build/Linux/XPlatCppLinux
+
+# Xbox build generated files
+build/Xbox/durango
```
Comment (Vicodex) [inline targets/XPlatCppSdk/source/.gitignore.ejs]: >d [](start = 11, length = 1)

I'd recommend to use [Dd] syntax (for all directories created by Visual Studio). I.e. [Dd]ebug, [Rr]elease, etc. Cases may change in some weird situations

File: targets/XPlatCppSdk/source/code/source/playfab/PlayFabIXHR2HttpPlugin.cpp [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+        auto response = postEventRequest.GetData();
+        reqContainer.responseString = std::string(response.begin(), response.end());
+
+        // 401 is a special case where the status does not bubble up and we have to parse it from the HRESULT
+        if (status == 401)
+        {
+            reqContainer.errorWrapper.HttpCode = 401;
+            reqContainer.errorWrapper.HttpStatus = "Access denied";
+
+            // TODO sangarg : Fix the error code here
```
Comment (sanchitgarg) [inline targets/XPlatCppSdk/source/code/source/playfab/PlayFabIXHR2HttpPlugin.cpp]: @Vicodex : Please check the error code here. Should we add a new error code for 401? We are currently using PlayFabErrorConnectionTimeout.

**Pull request merged**